ui/shared/PushHealthStatus.jsx

```diff
@@ -43,35 +48,51 @@ class PushHealthStatus extends Component {
     );
 
     if (!failureStatus) {
-      const { needInvestigation } = data;
-      const testsNeed = needInvestigation > 1 ? 'tests need' : 'test needs';
-      const healthStatus = needInvestigation
-        ? `${needInvestigation} ${testsNeed} investigation`
-        : 'OK';
-
-      this.setState({ healthStatus, needInvestigation });
+      this.setState({ ...data });
     }
   }
 
   render() {
     const { repoName, revision } = this.props;
-    const { healthStatus, needInvestigation } = this.state;
-    const color = needInvestigation ? 'danger' : 'success';
-    const extraTitle = needInvestigation ? 'Needs investigation' : 'Looks good';
+    const { needInvestigation, unsupported } = this.state;
+    const testsNeed = needInvestigation > 1 ? 'tests need' : 'test needs';
+    const icon =
+      needInvestigation + unsupported === 0 ? faCheck : faExclamationTriangle;
+    let healthStatus = 'OK';
+    let badgeColor = 'success';
+    let extraTitle = 'Looks good';
+
+    if (unsupported) {
+      healthStatus = `${unsupported} unsupported tests`;
+      badgeColor = 'warning';
+      extraTitle = 'Indeterminate';
+    }
+    if (needInvestigation) {
+      healthStatus = `${needInvestigation} ${testsNeed} investigation`;
+      badgeColor = 'danger';
+      extraTitle = 'Needs investigation';
+    }
 
     return (
-      <a
-        href={getPushHealthUrl({ repo: repoName, revision })}
-        target="_blank"
-        rel="noopener noreferrer"
-      >
-        <Badge
-          color={color}
-          title={`Push Health status - click for details: ${extraTitle}`}
-        >
-          {healthStatus}
-        </Badge>
-      </a>
+      <React.Fragment>
+        {needInvestigation !== null ? (
```

Comment:
I'm curious why you don't just use a falsey value (`!needInvestigation`).

Reply:
In this case, I'm distinguishing between `have no result yet` and `got 0 need investigation`.  If I use `!needInvestigation` it'll show "OK" by default, then if we get a result back with a number, it'll change to "Needs Investigation".  This will use the spinner till we get the result.